api/types/plugin.go

```diff
@@ -113,6 +117,16 @@ func (p *PluginV1) CheckAndSetDefaults() error {
 		if (bearer.Token == "") == (bearer.TokenFile == "") {
 			return trace.BadParameter("exactly one of Token and TokenFile must be specified")
 		}
+	case *PluginSpecV1_Jamf:
+		if settings.Jamf.JamfSpec.ApiEndpoint == "" {
+			return trace.BadParameter("api endpoint must be set")
+		}
+		if p.Credentials == nil {
+			return trace.BadParameter("credentials must be set")
+		}
+		if p.Credentials.GetIdSecret().Id == "" || p.Credentials.GetIdSecret().Secret == "" {
+			return trace.BadParameter("Jamf plugin requires Jamf account username and password")
```

Comment:
Maybe refer the same name as we store it. id and secret?

Reply:
`id` and `secret` is meant to be a [generic name](https://github.com/gravitational/teleport/blob/27eb75095ab5754177543d5f425415aac1dc51a3/api/proto/teleport/legacy/types/types.proto#L5117). In case of Jamf, that will be used as username and password. But for other oauth plugins client, it can be client id and secret. In this case, specifying username and password will help users (in the UI) to understand error context properly.